aws-rds-dbclusterparametergroup/src/main/java/software/amazon/rds/dbclusterparametergroup/BaseHandlerStd.java

```diff
@@ -304,53 +294,14 @@ private ProgressEvent<ResourceModel, CallbackContext> modifyParameters(final Pro
         return ProgressEvent.progress(model, context);
     }
 
-    protected boolean isDBClustersAvailable(final ProxyClient<RdsClient> proxyClient, final ResourceModel model) {
-        return proxyClient
-                .injectCredentialsAndInvokeIterableV2(Translator.describeDbClustersRequest(), proxyClient.client()::describeDBClustersPaginator)
-                .stream()
-                .flatMap(describeDbClustersResponse -> describeDbClustersResponse.dbClusters().stream())
-                .filter(dbCluster -> dbCluster.dbClusterParameterGroup().equalsIgnoreCase(model.getDBClusterParameterGroupName()))
-                .allMatch(dbCluster -> dbCluster.status().equals(AVAILABLE));
-    }
+    protected ProgressEvent<ResourceModel, CallbackContext> waitForStabilization(final ProgressEvent<ResourceModel, CallbackContext> progress) {
```

Comment:
Let's call it `sleep`:
```java
protected ProgressEvent<ResourceModel, CallbackContext> sleep(final ProgressEvent<ResourceModel, CallbackContext> progress, final Duration delay) {
    return ProgressEvent.defaultInProgressHandler(progress.getCallbackContext(), delay.getSeconds(), progress.getResourceModel());
}
```

Looking at this method, it makes me think it could go to Commons.handler and take its well-deserved spot there.

Next, let's move the conditional execution out of the method: what we got now is `execIfNot(...)`, and we are trying to flatten this out to `if (not()) { exec(); }`. Note that this conditional execution could be guarded by `Commons.execOnce` (see DBInstance and DBCluster code for the ref). This could be 1 way to address the comment I left above (concerning adding the new attribute to the `HandlerConfig` class).

Reply:
`Commons.execOnce` relay on `then` to change `callbackContext` flag.
`Commons.execOnce` does not work with progress that have delay because `then` only applied when the delay is 0.
Here what happened 
1.  excOnce called with inProgressDelay progress
2. `then` did not called.
3. Flag did not updated.
4. CFN enters same step again after delay time is spent.
5. code enters `execOnce`again as flag was not updated
6. endless Loop.

I proposed solution to update `execOnce` implementation to 
```
    public static <M, C> ProgressEvent<M, C> execOnce(
            final ProgressEvent<M, C> progress,
            final ProgressEventLambda<M, C> func,
            final Function<C, Boolean> conditionGetter,
            final VoidBiFunction<C, Boolean> conditionSetter
    ) {
        if (!conditionGetter.apply(progress.getCallbackContext())) {
            ProgressEvent<M, C> progressEvent =  func.enact();
            conditionSetter.apply(progressEvent.getCallbackContext(), true);
        }
        return progress;
    }
```
But @khebul  had deep concern that will not work in all execOnce scenarios specifically stabilization scenario.  [here](https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-rds/pull/292#discussion_r957604547) 

Here underlying implantation for more understanding
```
    public static <M, C> ProgressEvent<M, C> execOnce(
            final ProgressEvent<M, C> progress,
            final ProgressEventLambda<M, C> func,
            final Function<C, Boolean> conditionGetter,
            final VoidBiFunction<C, Boolean> conditionSetter
    ) {
        if (!conditionGetter.apply(progress.getCallbackContext())) {
            return func.enact().then(p -> {
                conditionSetter.apply(p.getCallbackContext(), true);
                return p;
            });
        }
        return progress;
    }
    //Underlying implemetation
    public ProgressEvent<ResourceT, CallbackT> then(Function<ProgressEvent<ResourceT, CallbackT>, ProgressEvent<ResourceT, CallbackT>> func) {
        return this.canContinueProgress() ? (ProgressEvent)func.apply(this) : this;
    }

    @JsonIgnore
    public boolean canContinueProgress() {
        return this.status == OperationStatus.IN_PROGRESS && this.callbackDelaySeconds == 0;
    }
```